apps/examples/src/examples/custom-grid/CustomGridExample.tsx

```diff
@@ -0,0 +1,108 @@
+import { useLayoutEffect, useRef } from 'react'
+import { TLComponents, Tldraw, approximately, useEditor, useIsDarkMode, useValue } from 'tldraw'
+import 'tldraw/tldraw.css'
+
+/**
+ * There's a guide at the bottom of this file!
+ */
+
+const components: TLComponents = {
+	// [1]
+	Grid: ({ size, ...camera }) => {
+		const editor = useEditor()
+
+		// [2]
+		const screenBounds = useValue('screenBounds', () => editor.getViewportScreenBounds(), [])
+		const devicePixelRatio = useValue('dpr', () => editor.getInstanceState().devicePixelRatio, [])
+		const isDarkMode = useIsDarkMode()
+
+		const canvas = useRef<HTMLCanvasElement>(null)
+
+		useLayoutEffect(() => {
```

Comment:
is `useLayoutEffect` necessary here? it seems like `useEffect` would suffice but lemme know if i'm missing something

Reply:
i think it is yeah. With `useEffect` the grid would lag behind the other canvas elements by one frame?

Reply:
I think that might be ok btw to be one frame behind. Esp. for slower machines, it makes me wonder if we should be encouraging this pattern of strong consistency vs. something that might make the rendering slower potentially.

Reply:
nah i think it's fine. drawing a few thousand lines on a canvas is so fast it's negligible.